docs/guides/data-loading.md

```diff
@@ -209,19 +207,18 @@ async function getLoaderData(productId: string) {
   return product;
 }
 
-export const loader: LoaderFunction = async ({
-  params,
-}) => {
-  return json<LoaderData>(
-    await getLoaderData(params.productId)
-  );
+export const loader = async ({ params }: LoaderArgs) => {
+  let LoaderData: Awaited<ReturnType<typeof getLoaderData>> | null = null
+  LoaderData = await getLoaderData(`${params.productId}`)
+
+  return json({product: LoaderData});
```

Comment:
is this all needed? could you try:

```ts
export const loader = async ({ params }: LoaderArgs) => {
  const product = await getLoaderData(params.productId);

  return json({  product });
```

and see if type inference works?

Reply:
yea thanks it's better. type inference work.